packages/dynamite/lib/src/openapi_builder.dart

```diff
@@ -1239,13 +1226,15 @@ TypeResult resolveObject(
                       extraJsonSerializableValues: extraJsonSerializableValues,
                     );
 
+                    final nullable = _isParameterNullable(
+                      schema.required?.contains(propertyName),
+                      null,
+                    );
+
                     b
                       ..name = _toDartName(propertyName)
                       ..type = refer(
-                        _makeNullable(
-                          result.name,
-                          !(schema.required ?? []).contains(propertyName),
```

Comment:
This ! inverts the contains statement, but you removed it

Reply:
Ok I must have been utterly incompetent when cherry picking this onto its separate branch.

This PR should only include the changes to `_toDartName` and not the other experimentations I did :)

Reply:
Ah no problem!